be/src/olap/base_compaction.cpp

```diff
@@ -16,513 +16,115 @@
+              << ", cumulative_base_ratio=" << cumulative_base_ratio
+              << ", interval_since_last_base_compaction=" << interval_since_last_base_compaction;
+    return OLAP_ERR_BE_NO_SUITABLE_VERSION;
 }
 
-bool BaseCompaction::_validate_need_merged_versions(
-        const vector<Version>& candidate_versions) {
-    if (candidate_versions.size() <= 1) {
-        LOG(WARNING) << "unenough versions need to be merged. size=" << candidate_versions.size();
-        return false;
-    }
-
-    // 1. validate versions in candidate_versions are continuous
-    // Skip the first element
-    for (unsigned int index = 1; index < candidate_versions.size(); ++index) {
-        Version previous_version = candidate_versions[index - 1];
-        Version current_version = candidate_versions[index];
-        if (current_version.first != previous_version.second + 1) {
-            LOG(WARNING) << "wrong need merged version. "
-                         << "previous_version=" << previous_version.first
-                         << "-" << previous_version.second
-                         << ", current_version=" << current_version.first
-                         << "-" << current_version.second;
-            return false;
-        }
-    }
-
-    // 2. validate m_new_base_version is OK
-    if (_new_base_version.first != 0
-            || _new_base_version.first != candidate_versions.begin()->first
-            || _new_base_version.second != candidate_versions.rbegin()->second) {
-        LOG(WARNING) << "new_base_version is wrong."
-                     << " new_base_version=" << _new_base_version.first
-                     << "-" << _new_base_version.second
-                     << ", vector_version=" << candidate_versions.begin()->first
-                     << "-" << candidate_versions.rbegin()->second;
-        return false;
-    }
-
-    VLOG(10) << "valid need merged version";
-    return true;
+const std::string& BaseCompaction::compaction_name() const {
+    return _compaction_name;
 }
 
-OLAPStatus BaseCompaction::_validate_delete_file_action() {
-    // 1. acquire the latest version to make sure all is right after deleting files
-    ReadLock rdlock(_tablet->get_header_lock_ptr());
-    const RowsetSharedPtr rowset = _tablet->rowset_with_max_version();
-    if (rowset == nullptr) {
-        LOG(INFO) << "version is -1 when validate_delete_file_action";
-        return OLAP_ERR_BE_ERROR_DELETE_ACTION;
-    }
-    Version spec_version = Version(0, rowset->end_version());
-    vector<RowsetReaderSharedPtr> rs_readers;
-    _tablet->capture_rs_readers(spec_version, &rs_readers);
-
-    if (rs_readers.empty()) {
-        LOG(INFO) << "acquire data sources failed. version="
-           << spec_version.first << "-" << spec_version.second;
-        return OLAP_ERR_BE_ERROR_DELETE_ACTION;
-    }
-
-    VLOG(3) << "delete file action is OK";
-
-    return OLAP_SUCCESS;
+ReaderType BaseCompaction::compaction_type() const {
+    return _compaction_type;
```

Comment:
there is no need to store it in class attribute, It's OK to just return a constant value.